anxcloud/struct_dns_records.go

```diff
@@ -0,0 +1,27 @@
+package anxcloud
+
+import "github.com/anexia-it/go-anxcloud/pkg/clouddns/zone"
+
+func flattenDnsRecords(records []zone.Record) []interface{} {
+	zoneRecords := []interface{}{}
```

Comment:
please use `make(..., 0, len(records))`

this allocates the array for the already known number of elements, preventing the code to do a lot of reallocations and moving. also, have a real type for the array entries, as it is a single known type. add a type alias for better readability.

```
type dnsRecordData map[string]interface{}
zoneRecords := make([]dnsRecordData, 0, len(records))
```